code/game/machinery/metaldetector.dm

```diff
@@ -0,0 +1,87 @@
+GLOBAL_LIST_INIT(metal_detector_items, typecacheof(list(
+	/obj/item/weapon/gun,
+	/obj/item/weapon/material,
+	/obj/item/weapon/melee,
+	/obj/item/device/transfer_valve,
+	/obj/item/weapon/grenade/,
+	/obj/item/ammo_casing/,
+	/obj/item/ammo_magazine
+	)))
+
+/obj/machinery/metal_detector
+	name = "metal detector"
+	desc = "An advanced metal detector used to detect weapons."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "metal_detector"
+	anchored = TRUE
+	req_one_access = list(access_security, access_heads)
+	idle_power_usage = 100
+
+	plane = MOB_PLANE
+	layer = ABOVE_MOB_LAYER
+
+	var/next_announcement_time	// so this doesn't get spammy.
+
+	var/alarm_delay = 2000
+
+/obj/machinery/metal_detector/New()
```

Comment:
This should be Initialize()

Reply:
I actually knew this one, I just forgot to push the commit for it. I'll batch all the fixes into one commit now.